book/src/framework/components/blockchains/solana.md

```diff
@@ -16,6 +16,14 @@ arm64 f4hrenh9it/solana:latest - used locally
   contracts_dir = "."
   # optional, in case you need some custom image
   # image = "solanalabs/solana:v1.18.26"
+
+# optional, in case you need to deploy some programs
+# this example assumes there is a mcm.so in the mounted contracts_dir directory
+# value is the program id to set for the deployed program instead of auto generating
+# useful for deterministic testing
+[blockchain_a.solana_programs]
+  mcm = "6UmMZr5MEqiKWD5jqTJd1WCR5kT8oZuFYBLJFi1o6GQX"
```

Comment:
do you think you might expand that topic a bit? for example:
* how should one load a program with auto-generated id?
* where should one get that program id from, if auto-generation is not desired?
* in what way is testing deterministic with static program id? what indeterminism there would be if I used different program ids for programs that are the same, i.e. with the same bytecode? or is program id derived from the bytecode somehow?

it's more than fine to just outline these topics in broad strokes and linking to resources which describe them in more details

Reply:
Thanks, 

> how should one load a program with auto-generated id?

then we do it the normal way, we just do [solana deploy program](https://solana.com/docs/programs/deploying#deploy-program) then the program will be deployed with its own program id.

> where should one get that program id from, if auto-generation is not desired?

basically just any public key, but usually the developer of the solana program will have one set , eg [here for mcm](https://github.com/smartcontractkit/chainlink-ccip/blob/609f7b0c9734b3cfc1d1a1aea0fd1ddf4c90bd0c/chains/solana/contracts/programs/mcm/src/lib.rs#L3-L3)

> in what way is testing deterministic with static program id? what indeterminism there would be if I used different program ids for programs that are the same, i.e. with the same bytecode? or is program id derived from the bytecode somehow?

basically during testing, we want to look up the deployed program based on the program id,
eg
```
data, accErr := s.SolanaClient.GetAccountInfoWithOpts(ctx, <PROGRAM_ID>, &rpc.GetAccountInfoOpts{
		Commitment: config.DefaultCommitment,
	})
```

If we dont tell solana to deploy a program with a certain program id, then we wont be able to look it up easily during testing since it is auto generated.

If we tell solana to deploy a program with a provided program id, then we can just look it up easily during testing.

Hope this make sense?

Reply:
I added more info  to the `solana.md` , let me know if that is enough or you prefer it to be in other places too